### DISCUSSION TITLE
What are the ways to trigger a detection on a camera so it creates an event

### DISCUSSION THREAD

**Discussion by NeonCandy** [blakeblackshear/frigate#7453] [Ask A Question] [answered]
I am running Frigate on Ubuntu Server without Home Assistant, all cameras are setup for 24/7 recording and have detection disabled.
My cameras support ONVIF and I managed to write a script that tells me if a camera is detecting motion.

Because I can't find Google Coral anywhere (or it's too expensive) I wanted to utilize that and trigger an event (detection) for that camera on Frigate so it shows up under events.

Im really new to all of this and am wondering how can I tell Frigate that a camera detected motion so it makes an event for it? When I was looking through their API I didn't see an endpoint for this, are the MQTT messages the only way to do this? Do I need an MQTT Broker or can I  send it straight to Frigate somehow? Also, do I need to have Detection enabled on a camera for this to work, because I don't want to waste CPU power on detection (that is what im tryin to avoid!).

**Comment by NickM-27** [comment, marked as answer]
> are the MQTT messages the only way to do this?

It is not possible with MQTT either. 

This feature has been implemented for 0.13

**Reply by NeonCandy** [reply]
So currently there is no way to accomplish this? Even with Home Assistant? I think I saw somewhere that you can do something similar with it

**Reply by NickM-27** [reply]
> So currently there is no way to accomplish this? Even with Home Assistant? I think I saw somewhere that you can do something similar with it

Correct, currently in frigate 0.12.1 there is no way to create an event manually

**Reply by NeonCandy** [reply]
Im on 0.13.0-7A2D09D, has it allready been implemented on this version or do I need a newer one?

**Reply by NickM-27** [reply]
Yes it's implemented there

**Reply by NeonCandy** [reply]
Thanks for the fast replies! Sorry to be bothering you like this, but is there some documentation on how exactly this is implemented and what I need to do to trigger a detection/event?

**Reply by NickM-27** [reply]
Yes, the preview docs linked at the top of the 0.13 PR

**Reply by NeonCandy** [reply]
Thanks! [This](https://deploy-preview-6262--frigate-docs.netlify.app/integrations/api#post-apieventscamera_namelabelcreate) is it right? Just wanted to post here in case anyone else stumbles on this thread.
Oh one more thing, do I need to have detection enabled on a camera in order for this event to register?

**Reply by NickM-27** [reply]
> Thanks! [This](https://deploy-preview-6262--frigate-docs.netlify.app/integrations/api#post-apieventscamera_namelabelcreate) is it right? Just wanted to post here in case anyone else stumbles on this thread.

Yes

> Oh one more thing, do I need to have detection enabled on a camera in order for this event to register?

No that is not required.

**Reply by NeonCandy** [reply]
Great! Thanks again for the help and fast replies :)

**Comment by scooper1** [comment]


**Reply by NickM-27** [reply]
That has nothing to do with creating motion events in Frigate
